packages/components/src/components/form/AutocompleteMultiSelect/AutocompleteMultiSelect.test.tsx

```diff
@@ -0,0 +1,252 @@
+    renderWithProviders(
+      <AutocompleteMultiSelect items={items}>
+        <AutocompleteMultiSelect.SelectedItems />
+        <AutocompleteMultiSelect.Input placeholder="Select an item" />
+        <AutocompleteMultiSelect.List renderItem={(item: string) => <div>{item}</div>} />
+      </AutocompleteMultiSelect>
+    );
+
+    const input = screen.getByPlaceholderText(/Select an item/);
+    fireEvent.focus(input);
+    fireEvent.change(input, { target: { value: 'foo' } });
+
+    const list = screen.getByRole('listbox');
+    fireEvent.click(list.firstElementChild);
+
+    let selectedItem = screen.getByText('foo');
+    expect(selectedItem).toBeInTheDocument();
+
+    const removeButton = selectedItem.parentElement.querySelector(
+      '[data-testid="cmpsr.autocompleteMultiSelect.clear-tag-button"]'
+    );
+    expect(removeButton).toBeInTheDocument();
+    fireEvent.click(removeButton);
+
+    expect(selectedItem).not.toBeInTheDocument();
+  });
+  test('should clear selected item when clear button is clicked', () => {
+    renderWithProviders(
+      <AutocompleteMultiSelect items={items}>
+        <AutocompleteMultiSelect.Input placeholder="AutocompleteMultiSelect" />
+        <AutocompleteMultiSelect.List renderItem={(item: string) => <div>{item}</div>} />
+      </AutocompleteMultiSelect>
+    );
+    const input = screen.getByPlaceholderText(/AutocompleteMultiSelect/);
+    fireEvent.focus(input);
+    fireEvent.change(input, { target: { value: 'foo' } });
+
+    const clearButton = screen.getByTestId('cmpsr.autocompleteMultiSelect.clear-button');
+    fireEvent.click(clearButton);
+    expect(input).toHaveValue('');
+  });
+  test('should render no results', () => {
+    renderWithProviders(
+      <AutocompleteMultiSelect items={[]}>
+        <AutocompleteMultiSelect.Input placeholder="AutocompleteMultiSelect" />
+        <AutocompleteMultiSelect.List renderItem={(item: string) => <div>{item}</div>} />
+      </AutocompleteMultiSelect>
+    );
+    const input = screen.getByPlaceholderText(/AutocompleteMultiSelect/);
+    fireEvent.focus(input);
+    const list = screen.getByRole('listbox');
+    expect(list.children).toHaveLength(1);
+    within(list).getByText(/No results/);
+  });
+  test('should render no results custom', () => {
+    renderWithProviders(
+      <AutocompleteMultiSelect items={[]}>
+        <AutocompleteMultiSelect.Input placeholder="AutocompleteMultiSelect" />
+        <AutocompleteMultiSelect.List
+          renderItem={(item: string) => <div>{item}</div>}
+          noResultsContent={<div>No results custom</div>}
+        />
+      </AutocompleteMultiSelect>
+    );
+    const input = screen.getByPlaceholderText(/AutocompleteMultiSelect/);
+    fireEvent.focus(input);
+    const list = screen.getByRole('listbox');
+    expect(list.children).toHaveLength(1);
+    within(list).getByText(/No results custom/);
+  });
+  test('should call onInputValueChange when input value changes', () => {
```

Comment:
Rename test description and mock to be consistent with the property `onSelectedItemsChange`

Reply:
done!